src/distilabel/tasks/text_generation/self_instruct.py

```diff
@@ -79,6 +90,97 @@ def generate_prompt(self, input: str) -> Prompt:
             formatted_prompt=self.template.render(**render_kwargs),
         )
 
+    @property
+    def output_args_names(self) -> List[str]:
+        return ["instructions"]
+
     def parse_output(self, output: str) -> Dict[str, List[str]]:
         """Parses the output of the model into the desired format."""
-        return {"generations": output.split("\n")}
+        pattern = re.compile(r"\d+\.\s+(.*?)\n")
+        print(pattern.findall(output))
+        return {"instructions": pattern.findall(output)}
+
+    def to_argilla_dataset(self, dataset_row: Dict[str, Any]) -> "FeedbackDataset":
```

Comment:
Should we check here if argilla is installed?

Reply:
No need to, as we're checking it in the `to_argilla` method already, and these methods are not intended to be used outside the `to_argilla` method of `CustomDataset`